dali/pipeline/operators/op_spec.h

```diff
@@ -416,6 +416,50 @@ class DLL_PUBLIC OpSpec {
   template <typename T, typename S>
   inline T GetArgumentImpl(const string &name, const ArgumentWorkspace *ws, Index idx) const;
 
+  /**
+   * @brief Check if the ArgumentInput of given shape can be used with GetArgument(),
+   *        representing a batch of scalars
+   *
+   * @argument should_throw whether this function should throw an error if the shape doesn't match
+   * @return true iff the shape is allowed to be used as Argument
+   */
+  bool CheckArgumentShape(const kernels::TensorListShape<> &shape, int batch_size,
+                          const std::string &name, bool should_throw = false) const {
+    if (shape.num_samples() == 1) {
+      // TODO(klecki): 1 sample version will be of no use after the switch to CPU ops unless we
+      // generalize accepted batch sizes throughout the pipeline
+      bool is_one_sample_with_batch = shape[0] == kernels::TensorShape<>(batch_size);
+      if (should_throw) {
+        DALI_ENFORCE(is_one_sample_with_batch,
+            "Unexpected shape of argument \"" + name +
+                "\". If only one tensor is passed, it should have a shape equal to {" +
+                std::to_string(batch_size) +
+                "}.  When accessing arguments as scalars 1 tensor of shape {" +
+                std::to_string(batch_size) + "} or " + std::to_string(batch_size) +
+                " tensors of shape {1} are expected. To access argument inputs where samples are "
```

Comment:
This `{1}` looks weird - made me look for some kind of formatting library.

Reply:
But that's the actual shape.

Reply:
I know - this just looks weird in the code. BTW - when we print TensorShape, we don't use braces - they are used in TensorListShape only; tensor shape would be printed as just `1`